lib/cosmos/gui/text/ruby_editor.rb

```diff
@@ -99,16 +99,16 @@ def self.create_format(color = Cosmos::BLACK, style='')
           ["/.*/", 0, STYLES['string']],
           # Double-quoted string, possibly containing escape sequences
           ['"([^"\\\\]|\\\\.)*"', 0, STYLES['string']],
-          # Match interpolated strings "blah #{code} blah"
-          ['(\#\\{)[^\\}]*\\}', 1, STYLES['brace']],
-          ['\#\\{[^\\}]*(\\})', 1, STYLES['brace']],
-          ['\#\\{([^\\}]*)\\}', 1, STYLES['normal']],
           # Single-quoted string, possibly containing escape sequences
           ["'[^'\\\\]*(\\.[^'\\\\]*)*'", 0, STYLES['string']],
           # Back-tick string, possibly containing escape sequences
           ["`[^`\\\\]*(\\.[^`\\\\]*)*`", 0, STYLES['string']],
+          # Match interpolated strings "blah #{code} blah"
+          ['(\#\\{)[^\\}]*\\}', 1, STYLES['brace']],
+          ['\#\\{[^\\}]*(\\})', 1, STYLES['brace']],
+          ['\#\\{([^\\}]*)\\}', 1, STYLES['normal']],
           # A single # possibly followed by matched quotes
-          ['#(?=([^\'"]*(\'|")[^\'"]*(\'|"))*[^\'"]*$).*', 0, STYLES['comment']],
+          ['#(?=([^\'")]*(\'|")[^\'"]*\2)*$).*', 0, STYLES['comment']],
       ])
```

Comment:
This is crazy because it works in Rubular and I think it's correct but it doesn't work in Script Runner. It misses the basic case of a single comment line.